src/utilities.cpp

```diff
@@ -420,4 +420,10 @@ ModelPtr parentModel(const EntityPtr &entity)
     return model;
 }
 
+void removeComponentFromEntity(const EntityPtr &entity, const ComponentPtr &component)
+{
+    auto componentEntity = std::dynamic_pointer_cast<ComponentEntity>(entity);
+    componentEntity->removeComponent(component, false);
```

Comment:
Do we not need to worry about the cast failing because this is internal code so we know it will only be called with a `ComponentEntity` entity?

Reply:
That's the idea, but if you can come up with a test that shows the opposite then please do.